doc/design/effects_mutability.md

```diff
@@ -454,6 +603,32 @@ Precise effects typing of callback functions is useful for local optimization ev
 intermediary (e.g. `iter2`) is not specialized to take advantage of the precision. And if the entire
 call chain is inlined, the precise effect typing enables optimal machine code generation.
 
+### Explicit concealed effects
+
+Partial application requires closure allocation. Functions like `Uns.( + )` do not require
+allocation, and because addition is critical functionality in low-level code, the API is explicit
+regarding (lack of) concealed effects.
+
+```hemlock
+    val ( + ): uns --> uns --> uns
+```
+
+However, if `+` is partially applied, the compiler transforms the function to one with an `alloc`
+effect, with signature equivalent to `+*`.
+
+```hemlock
+    val ( +* ): uns -{alloc}-> (uns --> uns)
+```
+
+As a consequence, partial application is prohibited in functions like `f`, and allowed in functions
+like `g` and `h`.
+
+```hemlock
+    val f: uns --> uns --> uns
+    val g: uns -{alloc}-> (uns --> uns)
+    val h: uns --> uns -{alloc}-> uns
```

Comment:
I don't understand how `h` works. It isn't possible to close on one parameter, line `h 1` since doing so would require a concealed `alloc` effect. If we provide two parameters, `h 1 2`, no closure is needed so the concealed `alloc` is superfluous. What am I missing?

Reply:
Actually, I misread this. I disregarded the statement immediately above 

> However, if `+` is partially applied, the compiler transforms the function to one with an `alloc`
> effect, with signature equivalent to `+*`.
>
>```hemlock
>    val ( +* ): uns -{alloc}-> (uns --> uns)
>```